microsoft/testsuites/storage/performance.py

```diff
@@ -0,0 +1,189 @@
+                data_disk_iops=5000,
+                data_disk_count=search_space.IntRange(min=16),
+            ),
+        ),
+    )
+    def perf_premium_datadisks_4k(self, node: Node, environment: Environment) -> None:
+        self._perf_premium_datadisks(node, environment)
+
+    @TestCaseMetadata(
+        description="""
+        This test case uses fio to test data disk performance using 1024K block size.
+        """,
+        priority=2,
+        timeout=TIME_OUT,
+        requirement=simple_requirement(
+            min_core_count=72,
+            disk=schema.DiskOptionSettings(
+                disk_type=schema.DiskType.PremiumSSDLRS,
+                data_disk_iops=5000,
+                data_disk_count=search_space.IntRange(min=16),
+            ),
+        ),
+    )
+    def perf_premium_datadisks_1024k(
+        self, node: Node, environment: Environment
+    ) -> None:
+        self._perf_premium_datadisks(node, environment, block_size=1024)
+
+    def _perf_premium_datadisks(
+        self, node: Node, environment: Environment, block_size: int = 4
+    ) -> None:
+        disk = node.features[Disk]
+        data_disks = disk.get_raw_data_disks()
+        disk_count = len(data_disks)
+        assert_that(disk_count).described_as(
+            "At least 1 data disk for fio testing."
+        ).is_greater_than(0)
+        partition_disks = _format_disk(node, data_disks)
+        print(partition_disks)
+        _stop_raid(node)
+        _make_raid(node, partition_disks)
+        filename = "/dev/md0"
+        cpu = node.tools[Lscpu]
+        core_count = cpu.get_core_count()
+        start_qdepth = 1
+        max_qdepth = 1024
+        numjobiterator = 0
+        fio_result_list: List[FIOResult] = []
+        fio = node.tools[Fio]
+        num_jobs = [1, 1, 2, 2, 4, 4, 8, 8, 8, 16, 16, 16]
+        for mode in FIOMODES:
+            qdepth = start_qdepth
+            numjobindex = 0
+            while qdepth <= max_qdepth:
+                numjob = num_jobs[numjobindex]
+                iodepth = int(qdepth / numjob)
+                fio_result = fio.launch(
+                    name=f"iteration{numjobiterator}",
+                    filename=filename,
+                    time=20,
+                    size_gb=1024,
+                    block_size=f"{block_size}K",
+                    mode=mode.name,
+                    gtod_reduce=False,
+                    iodepth=int(iodepth),
+                    numjob=numjob,
+                    overwrite=True,
+                )
+                fio_result_list.append(fio_result)
+                qdepth = qdepth * 2
+                numjobindex += 1
+                numjobiterator += 1
+        fio_messages: List[FIOResultsMessage] = fio.parse_test_results(fio_result_list)
+        self._handle_and_send_back_results(
+            core_count, disk_count, environment, fio_messages, block_size
+        )
+
+    def _handle_and_send_back_results(
```

Comment:
this method is sharable. Please find a way to share it.

Reply:
updated